File: Class/1.rb
class Cal
    def initialize(v1, v2) # 생성자(Constructor) 새로운 인스턴스가 생기면 반드시 실행됨! (=초기화)
        @v1 = v1 # 앞에 @가 붙으면 인스턴스 변수. 즉, 해당 Class 안의 모든 메소드에서 사용됨
        @v2 = v2
        # @가 붙지 않으면 지역 변수. 즉 initialize 안에서만 사용 가능
    end        

    def add()
        return @v1 + @v2
    end
    
    def subtract()
        return @v1 - @v2
    end
end

c1 = Cal.new(10,10) #c1 라는 인스턴스
p c1.add()
p c1.subtract()

c2 = Cal.new(30,20) #c2 라는 인스턴스
p c2.add()
p c2.subtract()

File: Class/1.py
class Cal(object): # object는 써도 되고 안 써도 됨. 상속과 관련된 것
    def __init__(self, v1, v2): # 첫 번째 매개 변수(self, 이름 바꿔도 됨)는 꼭 지정해야 함! 인스턴스 변수 지정 시 앞에 붙음
        self.v1 = v1
        self.v2 = v2

    def add(self): #self를 기입해야 인스턴스 변수를 사용할 수 있음
        return self.v1 + self.v2
        
    def subtract(self):
        return self.v1 - self.v2

c1 = Cal(10,10)
print(c1.add())
print(c1.subtract())

c2 = Cal(30,20)
print(c2.add())
print(c2.subtract())